Packs/Box/Integrations/BoxV2/BoxV2.py

```diff
@@ -0,0 +1,2025 @@
+        readable_output=readable_output
+    )
+
+
+def download_file_command(auth_params: dict, base_url: str, verify: bool, proxy: bool,
+                          args: Dict[str, Any]) -> dict:
+    """
+    Command executed when `box-download-file` is called. Will download a file based on the given
+    arguments.
+
+    This function requires a new JWT grant to be created with the explicit permissions of the user
+    who is downloading the file. This is why we are throwing away the main client and building a new
+    one here.
+
+    :param proxy: Indicates if using a proxy.
+    :param verify: Indicates if the client will verify self signed certs
+    :param base_url: Base URL for the service
+    :param auth_params: Args used for the authentication
+    :param args: demisto.args() dictionary - Used to pass the necessary arguments to the client
+    function.
+    :return: CommandResults - Returns a CommandResults object which is consumed by the
+    return_results function in main()
+    """
+    file_id: str = args.get('file_id')  # type:ignore
+    as_user: str = args.get('as_user')  # type:ignore
+
+    download_client = Client(
+        auth_params=auth_params,
+        base_url=base_url,
+        verify=verify,
+        proxy=proxy,
+        as_user=as_user
+    )
+
+    response = download_client.download_file(file_id=file_id)
+
+    d = response.headers['content-disposition']
+    file_name = get_filename(d)
+    return fileResult(filename=file_name, data=response.content)
+
+
+''' MAIN FUNCTION '''
+
+
+def test_module(client: Client) -> str:
+    """
+    This test assumes that the account has at least one user. By definition, if the instance is
+    configured correctly, there will always be a minimum of one user.
+
+    :param client:
+    :return:
+    """
+    response: Response = client.list_users(limit=1)
+    if response:
+        return 'ok'
+    else:
+        return 'An error occurred.'
+
+
+def fetch_incidents(client: Client, max_results: int, last_run: dict, first_fetch_time: int,
+                    as_user: str) -> Tuple[str, List[Dict]]:
+    """
+
+    :param client:
+    :param max_results:
+    :param last_run:
+    :param first_fetch_time:
+    :param as_user:
+    :return:
+    """
+    created_after = last_run.get('time', None)
+    incidents = []
+    if not created_after:
+        created_after = datetime.fromtimestamp(first_fetch_time, tz=timezone.utc).strftime(
+            DATE_FORMAT)
+    results = client.list_events(stream_type='admin_logs', as_user=as_user, limit=max_results,
+                                 created_after=created_after)
+    raw_incidents = results.get('entries', [])
+    next_run = datetime.now(tz=timezone.utc).strftime(DATE_FORMAT)
+    for raw_incident in raw_incidents:
+        event = Event(raw_input=raw_incident)
+        xsoar_incident = event.format_incident()
+        incidents.append(xsoar_incident)
+        if event.created_at > created_after:
```

Comment:
are you sure you will get the latest to be created as last in the loop? if not, you need to make sure you save the latest to be created in the chunk you got.

Reply:
In this case yes. Box returns a sorted list of events with the create time going from earliest to latest.